src/index.js

```diff
@@ -10,12 +10,15 @@ async function action() {
 
   const buildParameters = await BuildParameters.create();
   const baseImage = new ImageTag(buildParameters);
-
-  // Build docker image
-  const builtImage = await Docker.build({ path: actionFolder, dockerfile, baseImage });
-
-  // Run docker image
-  await Docker.run(builtImage, { workspace, ...buildParameters });
+  if (buildParameters.kubernetes) {
+    core.info('Building with Kubernetes');
+    await Kubernetes.runBuildJob(buildParameters, baseImage);
+  } else {
+    // Build docker image
+    // TODO: No image required (instead use a version published to dockerhub for the action, supply credentials for github cloning)
+    const builtImage = await Docker.build({ path: actionFolder, dockerfile, baseImage });
```

Comment:
Should we create an issue for this todo instead? I think it's out of scope?

I'm also curious why you would want this thin wrapper to be published at all.

Reply:
Ah, so what this was saying is. If the base image is published to dockerhub and then used within the action as a base, it would make the kubernetes container process a little nicer. Instead of needing to clone the webbertekken repository into the kubernetes container to access the shell scripts in the 'steps' directory, a base image could be supplied that includes them. 

I suppose it would save a small amount of time when not using kubernetes also, as the base layer wouldn't be built every time.

Reply:
Right. We should create a ticket for this as it's one of the most important long-term improvements, which would possibly also allow us to become CI agnostic.

(as the biggest challenge is dealing with different docker versions from @GabLeRoux from within a Github Action, which may be solved by DinD and in which case we might wrap and combine the whole javascript part with it too - ultimately that would condone moving some changes from this project to the unity3d docker project)